src/Mod/Sketcher/Gui/TaskSketcherElements.cpp

```diff
@@ -295,54 +383,141 @@ TaskSketcherElements::TaskSketcherElements(ViewProviderSketch *sketchView)
+}
 
-    connectionElementsChanged.disconnect();
+void TaskSketcherElements::on_listMultiFilter_itemChanged(QListWidgetItem* item)
+{
+    ui->listMultiFilter->blockSignals(true);
+
+    if (item == ui->listMultiFilter->item(3)) { //3 is 'All geos'
+        for (int i = 4; i < ui->listMultiFilter->count(); i++) {
+            ui->listMultiFilter->item(i)->setCheckState(item->checkState());
+        }
+    }
+
+    ui->listMultiFilter->blockSignals(false);
+
+    updateVisibility();
+}
+
+void TaskSketcherElements::setItemVisibility(QListWidgetItem* item)
+{
+    /* index
+    0 => Normal
+    1 => Construction
+    2 => External
+    3 => all geos
+    4 => Point
+    5 => Line
+    6 => Circle
+    7 => Ellipse
+    8 => Arc
+    9 => Arc of Ellipse
+    10 => Hyperbola
+    11 => Parabola
+    12 => bspline
+    */
+
+    ElementData* itemData = qvariant_cast<ElementData*>(item->data(Qt::UserRole));
+    bool visibility = true;
+
+    if (ui->filterBox->checkState() == Qt::Unchecked) { item->setHidden(false); return; }
+
+    if (ui->listMultiFilter->item(0)->checkState() == Qt::Unchecked && !itemData->isConstruction) { item->setHidden(true); return; }
+    if (ui->listMultiFilter->item(1)->checkState() == Qt::Unchecked && itemData->isConstruction) { item->setHidden(true); return; }
+    if (ui->listMultiFilter->item(2)->checkState() == Qt::Unchecked && itemData->isExternal) { item->setHidden(true); return; }
+
+    if (ui->listMultiFilter->item(4)->checkState() == Qt::Unchecked && itemData->GeometryType == Part::GeomPoint::getClassTypeId()) { item->setHidden(true); return; }
+    if (ui->listMultiFilter->item(5)->checkState() == Qt::Unchecked && itemData->GeometryType == Part::GeomLineSegment::getClassTypeId()) { item->setHidden(true); return; }
+    if (ui->listMultiFilter->item(6)->checkState() == Qt::Unchecked && itemData->GeometryType == Part::GeomCircle::getClassTypeId()) { item->setHidden(true); return; }
+    if (ui->listMultiFilter->item(7)->checkState() == Qt::Unchecked && itemData->GeometryType == Part::GeomEllipse::getClassTypeId()) { item->setHidden(true); return; }
+    if (ui->listMultiFilter->item(8)->checkState() == Qt::Unchecked && itemData->GeometryType == Part::GeomArcOfCircle::getClassTypeId()) { item->setHidden(true); return; }
+    if (ui->listMultiFilter->item(9)->checkState() == Qt::Unchecked && itemData->GeometryType == Part::GeomArcOfEllipse::getClassTypeId()) { item->setHidden(true); return; }
+    if (ui->listMultiFilter->item(10)->checkState() == Qt::Unchecked && itemData->GeometryType == Part::GeomArcOfHyperbola::getClassTypeId()) { item->setHidden(true); return; }
+    if (ui->listMultiFilter->item(11)->checkState() == Qt::Unchecked && itemData->GeometryType == Part::GeomArcOfParabola::getClassTypeId()) { item->setHidden(true); return; }
+    if (ui->listMultiFilter->item(12)->checkState() == Qt::Unchecked && itemData->GeometryType == Part::GeomBSplineCurve::getClassTypeId()) { item->setHidden(true); return; }

```

Comment:
Maybe there is better way to present it. Such as an if with multiline ORed conditions. So `{ item->setHidden(true); return; }` isn't indefinitely repeated.

Reply:
I don't know what is a if with multiline ORed conditions. Can you please suggest modification?

Reply:
```
    if (    (ui->listMultiFilter->item(0)->checkState() == Qt::Unchecked && !itemData->isConstruction) ||
            (ui->listMultiFilter->item(1)->checkState() == Qt::Unchecked && itemData->isConstruction) ||
            (ui->listMultiFilter->item(2)->checkState() == Qt::Unchecked && itemData->isExternal) ||
            (ui->listMultiFilter->item(4)->checkState() == Qt::Unchecked && itemData->GeometryType == Part::GeomPoint::getClassTypeId()) ||
            (ui->listMultiFilter->item(5)->checkState() == Qt::Unchecked && itemData->GeometryType == Part::GeomLineSegment::getClassTypeId()) ||
            (ui->listMultiFilter->item(6)->checkState() == Qt::Unchecked && itemData->GeometryType == Part::GeomCircle::getClassTypeId()) ||
            (ui->listMultiFilter->item(7)->checkState() == Qt::Unchecked && itemData->GeometryType == Part::GeomEllipse::getClassTypeId()) ||
            (ui->listMultiFilter->item(8)->checkState() == Qt::Unchecked && itemData->GeometryType == Part::GeomArcOfCircle::getClassTypeId()) ||
            (ui->listMultiFilter->item(9)->checkState() == Qt::Unchecked && itemData->GeometryType == Part::GeomArcOfEllipse::getClassTypeId()) ||
            (ui->listMultiFilter->item(10)->checkState() == Qt::Unchecked && itemData->GeometryType == Part::GeomArcOfHyperbola::getClassTypeId()) ||
            (ui->listMultiFilter->item(11)->checkState() == Qt::Unchecked && itemData->GeometryType == Part::GeomArcOfParabola::getClassTypeId()) ||
            (ui->listMultiFilter->item(12)->checkState() == Qt::Unchecked && itemData->GeometryType == Part::GeomBSplineCurve::getClassTypeId())
            ) {
        item->setHidden(true);
        return;
    }
```

Reply:
Oh yes with or. I see now.